From 4e6eb9a5f556458e745e956de43d2f00f7a94c94 Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Tue, 15 May 2012 17:30:30 +0300
Subject: [PATCH 181/311] conf/buf.c: replace c2_alloc() with C2_ALLOC_ARR()

C2_ALLOC_ARR() can (it doesn't at the moment) issue a more meaningful
error message.

LogD: #1321
RB: r/714
---
 conf/buf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/conf/buf.c b/conf/buf.c
index 2976ce9..9d08782 100644
--- a/conf/buf.c
+++ b/conf/buf.c
@@ -33,7 +33,7 @@ bool c2_conf_buf_eq(const struct c2_conf_buf *x, const struct c2_conf_buf *y)
 int c2_conf_buf_copy(struct c2_conf_buf *dest, const struct c2_conf_buf *src)
 {
 	C2_PRE(dest->cb_size == 0 && dest->cb_data == NULL);
-	dest->cb_data = c2_alloc(src->cb_size);
+	C2_ALLOC_ARR(dest->cb_data, src->cb_size);
 	if (dest->cb_data == NULL)
 		return -ENOMEM;
 	dest->cb_size = src->cb_size;
-- 
1.8.3.2

